enhancements/ingress/configurable-dns-loadbalancerservice.md

```diff
@@ -0,0 +1,343 @@
+  // <snip>
+
+  // dnsManagementPolicy denotes the current policy applied on the dns record.
+  // Records that have policy set as "Unmanaged" are ignored by the ingress
+  // operator and the DNS record on the cloud provider is not managed by the
+  // operator. This record on the cloud provider can be deleted at the
+  // discretion of the cluster admin.
+  //
+  // +kubebuilder:default:="Managed"
+  // +kubebuilder:validation:Optional
+  // +kubebuilder:validation:Enum=Managed;Unmanaged
+  DNSManagementPolicy DNSManagementPolicy `json:"dnsManagementPolicy"`
+}
+
+// DNSManagementPolicy is a policy for configuring how the dns controller 
+// manages DNSRecords.
+//
+// +kubebuilder:validation:Enum=Managed;Unmanaged
+type DNSManagementPolicy string
+
+const (
+    // ManagedDNS configures the dns controller manage the lifecyle of the
+    // dns record on the appropriate platform.
+    ManagedDNS DNSManagementPolicy = "Managed"
+    // UnmanagedDNS configures the dns controller to ignore the dns record
+    // and allows the DNS record on the cloud provider to be managed
+    // by the cluster admin.
+    UnmanagedDNS DNSManagementPolicy = "Unmanaged"
+)
+```
+
+Re-uses existing condition `DNSManaged` of the controller to denote if
+the wildcard *DNSRecord* associated with the controller is managed/unmanaged. 
+
+### Implementation Details/Notes/Constraints
+
+Based on `.loadbalancer.dnsManagementPolicy`, the controller decides whether to ensure creating
+a wildcard *DNSRecord*. If the customer is updating `.loadbalancer.dnsManagementPolicy` from 
+`"Managed"` to `"Unmanaged"` the previously created *DNSRecord* will be updated
+appropriately.
+
+The existing `.loadbalancer.dnsManagementPolicy` value of the ingresscontroller
+will be replicated to the *DNSRecord*'s `.spec.dnsManagementPolicy` as well, to
+clearly state which dns record is unmanaged. The cluster admin can choose to delete
+the *DNSRecord* at his own discretion (this will involve also manually removing the finalizer).
+
+Once the *DNSRecord* is set to `Unmanaged`, any updates done to it will be ignored
+by the operator, but it is to be noted that if the ingresscontroller is updated from
+`Unmanaged` to `Managed` any changes done will be lost during re-sync.
+
+If the ingresscontroller is deleted when `.loadbalancer.dnsManagementPolicy` is set to
+`Unmanaged`, the *DNSRecord* CR will be deleted as per current implementation but the
+DNS record on the cloud provider will not be deleted by the operator and will require
+the cluster admin to manually delete it. 
+
+The support provided by the installer and this new field aren't entirely
+mutually exclusive, i.e. if customer has opted to disable DNS management cluster
+wide (as documented
+[here](https://github.com/openshift/installer/blob/master/docs/user/aws/install_upi.md#remove-dns-zones)),
+this field is of no value and need not be set since *DNSRecord* created is silently
+reconciled by the DNS controller.
```

Comment:
```suggestion
this field is of no value and need not be set since reconciliation of the *DNSRecord*
CR is a no-op when no DNS zones are configured.
```